Form/Transformer/BooleanTypeToBooleanTransformer.php

```diff
@@ -0,0 +1,56 @@
+<?php
+
+/*
+ * This file is part of the FOSRestBundle package.
+ *
+ * (c) FriendsOfSymfony <http://friendsofsymfony.github.com/>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace FOS\RestBundle\Form\Transformer;
+
+use Symfony\Component\Form\DataTransformerInterface;
+use Symfony\Component\Form\Exception\TransformationFailedException;
+
+/**
+ * @author GeLo <geloen.eric@gmail.com>
+ */
+class BooleanTypeToBooleanTransformer implements DataTransformerInterface
+{
+    /**
+     * {@inheritdoc}
+     */
+    public function transform($value)
+    {
+        if ($value === null) {
+            return;
+        }
+
+        if (!is_bool($value)) {
+            throw new TransformationFailedException(sprintf(
+                'The boolean type expects a boolean or null value, got "%s"',
+                is_object($value) ? get_class($value) : gettype($value)
+            ));
+        }
+
+        return $value;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function reverseTransform($value)
+    {
+        $filterValue = !is_object($value)
+            ? filter_var($value, FILTER_VALIDATE_BOOLEAN, array('flags' => FILTER_NULL_ON_FAILURE))
+            : $value;
```

Comment:
I'm not really fan of this block but it seems `filter_var` is buggy with object.

Quoting the doc:

> Returns TRUE for "1", "true", "on" and "yes". Returns FALSE otherwise.
>
> If FILTER_NULL_ON_FAILURE is set, FALSE is returned only for "0", "false", "off", "no", and "", and NULL is returned for all non-boolean values.

Unfortunatelly, passing an `\stdClass` (as it is done in the tests) returns `false` instead of `null`. So I workaround it this way for now...